### PR TITLE
Refactoring of rad_profile module

### PR DESCRIPTION
**Pull request by nollety**
# Description

Refactored `eradiate.radprops.rad_profile` module so that radiative property profiles can be instanciated from a thermophysical properties data set, instead of (previously) from parameters used for generating thermophysical properties data set.

:warning: This pull request is a prerequisite for #102.

This pull request should be merged **after** #111 (this will probably requires some alignment work).

## Example

Create a `type="afgl_1986"` radiative property profile using:

```python
import numpy as np

from eradiate.units import unit_registry as ureg
from eradiate.radprops.rad_profile import AFGL1986RadProfile
from eradiate.thermoprops.afgl_1986 import make_profile

thermoprops = make_profile(
    model_id="us_standard",
    levels=np.linspace(0, 120, 121) * ureg.km,
    concentrations={"CO2": 400 * ureg.dimensionless}
)
AFGL1986RadProfile(thermoprops=thermoprops)
```

or, equivalently:

```python
AFGL1986RadProfile(
    thermoprops=dict(
        model_id="us_standard",
        levels=np.linspace(0, 120, 121) * ureg.km,
        concentrations={"CO2": 400 * ureg.dimensionless},
    )
)
```

## Changes

* Removed `levels` attribute from `US76ApproxRadProfile`
* Removed `model`, `levels` and `concentrations` attributes from `AFGL1986RadProfile`
* Added `thermoprops` attribute to `US76ApproxRadProfile` and `AFGL1986RadProfile`
* Renamed RadProfile's methods that have a `spectral_ctx` argument:
  * `sigma_a()` -> `eval_sigma_a()`
  * `sigma_s()` -> `eval_sigma_s()`
  * `sigma_t()` -> `eval_sigma_t()`
  * `albedo()` -> `eval_albedo()`
* updated `ArrayRadProfile` to support 1D profiles only for consistency
* Added a `from_thermoprops_params` classmethod to `US76ApproxRadProfile` and `AFGL1986RadProfile`: these constructors forward their `model_id`, `levels` and `concentrations`  parameters to the functions that generate the corresponding thermophysical properties data set.
* updated `eradiate.thermoprops.afgl1986.make_profile()` to accept `levels` and `concentrations` parameters
* Added type annotations
* Updated docstrings
* Updated/added tests:
  * updated `rad_profile` tests according to the changes in `RadProfile` classes
  * added test cases for the module `thermoprops.afgl1986`

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Review by leroyvn — CHANGES_REQUESTED** [review]
Hi @nollety, thanks for this PR :) It's fine overall but I'd like to switch from a constructor-based to a converter-based approach for the `from_thermoprops_params()` method.

File: eradiate/radprops/rad_profile.py
Comment (nollety) [reply eradiate/radprops/rad_profile.py:85]: Yes, I ignore the divide-by-zero warnings. Somehow, the warnings are still there even if I use `np.where`...

File: eradiate/radprops/rad_profile.py
Comment (nollety) [reply eradiate/radprops/rad_profile.py:888]: Oh yes, I like the converter idea much better!
I did not think about using a converter. I'll refactor `US76ApproxRadProfile` and `AFGL1986RadProfile` classes to use such converters.

These are the only two classes that have a `_thermoprops` attribute, by the way. The interface (`RadProfile`) does not define a `_thermoprops` attribute or `thermoprops` property like `US76ApproxRadProfile` and `AFGL1986RadProfile` do. Or did I misunderstand your last comment?

**Comment by nollety** [comment]
I'm waiting for your approval to resolve the two conversations above @leroyvn :)
I'll mark this pull request as draft until #111 is merged at which point I'll come back to rebase/refactor and squash

File: eradiate/radprops/rad_profile.py
Comment (nollety) [reply eradiate/radprops/rad_profile.py:85]: np.divide(sigma_s, sigma_t, where=sigma_t!= 0.0, out=np.zeros_like(sigma_s))

**Comment by nollety** [comment]
@leroyvn I've squashed the commits and rebased onto the latest main; this is ready for review/merge !
I've marked the conversations above as resolved, I hope this is fine with you :)

**Comment by leroyvn** [comment]
Merging, thanks @nollety!

**Pull request merged**
